src/components/BoxLabel/BoxLabel.stories.tsx

```diff
@@ -1,13 +1,77 @@
 import { Meta, Story } from "@storybook/react"
-import { BoxLabel as BoxLabelComponent } from "./index"
+import { BoxLabel as BoxLabelComponent, BoxLabelProps } from "./index"
+import { IconMap } from "../../types"
+import { FiAlertCircle, IoSparklesSharp } from "react-icons/all"
+import React from "react"
+import { Icon } from "@chakra-ui/react"
 
-const Template: Story = () => {
-  return <BoxLabelComponent>Simple Box Label</BoxLabelComponent>
+interface BoxLabelStoryProps extends BoxLabelProps {}
+
+const iconMap: IconMap = {
+  Sparkle: <Icon as={IoSparklesSharp} />,
+  Alert: <Icon as={FiAlertCircle} />,
+}
+
+const Template: Story<BoxLabelStoryProps> = (args) => {
+  return (
+    // @ts-ignore
+    <BoxLabelComponent {...args} icon={iconMap[args.icon]}>
+      Simple Box Label
+    </BoxLabelComponent>
+  )
 }
 
 export const BoxLabel = Template.bind({})
 
 export default {
   title: "BoxLabel",
   component: BoxLabel,
+  argTypes: {
+    icon: {
+      control: {
+        type: "select",
+        options: {
+          Undefined: undefined,
+          Sparkle: "Sparkle",
+          Alert: "Alert",
+        },
+      },
+    },
```

Comment:
Can we remove the `ts-ignore` by changin the way we pass icon from args to BoxLabel?

```suggestion
const Template: Story<BoxLabelStoryProps> = (args) => {
  return (
    <BoxLabelComponent {...args} icon={args.icon}>
      Simple Box Label
    </BoxLabelComponent>
  )
}

export const BoxLabel = Template.bind({})

export default {
  title: "BoxLabel",
  component: BoxLabel,
  argTypes: {
    icon: {
      control: {
        type: "select",
        options: {
          Undefined: undefined,
          Sparkle: iconMap.Sparkle,
          Alert: iconMap.Alert,
        },
      },
    },
```